esprima.js

```diff
@@ -1274,27 +1281,13 @@
 
                     default:
                         if (isOctalDigit(ch)) {
```

Comment:
This is what you the consolidated code already does. Considering the rarity of this path, I think the consolidation clarity trumps the optimization.